extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/BeanArchiveProcessor.java

```diff
@@ -130,14 +131,28 @@ private IndexView buildApplicationIndex(ArcConfig config, ApplicationArchivesBui
         return CompositeIndex.create(indexes);
     }
 
-    private boolean isApplicationArchiveExcluded(ArcConfig config, ApplicationArchive archive) {
+    private boolean isApplicationArchiveExcluded(ArcConfig config, List<ExcludeDependencyBuildItem> excludeDependencyBuildItems,
+            ApplicationArchive archive) {
         if (archive.getArtifactKey() != null) {
             AppArtifactKey key = archive.getArtifactKey();
             for (IndexDependencyConfig excludeDependency : config.excludeDependency.values()) {
                 if (Objects.equal(key.getArtifactId(), excludeDependency.artifactId)
                         && Objects.equal(key.getGroupId(), excludeDependency.groupId)) {
-                    if (excludeDependency.classifier.isPresent()) {
-                        return Objects.equal(key.getClassifier(), excludeDependency.classifier.get());
+                    if (excludeDependency.classifier.isPresent()
+                            && Objects.equal(key.getClassifier(), excludeDependency.classifier.get())) {
```

Comment:
I'd rather use the original version...
```java
 if (excludeDependency.classifier.isPresent()) {
    return Objects.equal(key.getClassifier(), excludeDependency.classifier.get());
```

Reply:
Well, with the original version I would miss some false, I need to check false if they are true for the new BuildItem.

Reply:
Hm, it might make sense to add a util method to match and archive:
```java
static boolean archiveMatches(ApplicationArchive archive, String artifactId, String groupId, String classifier) {
  ...
}
```
And a simple test ;-).

Reply:
Done.

I would like to test the added build item, but I don't understand well the philosophy or the tests.

How can I create a test that call the build method passing all the needed build items ?

Can you pinpoint me to an example ?

Reply:
No need to test the build item (in fact, I don't think there's an easy way to add a dependency for a specific `QuarkusUnitTest`). Just add a regular unit test for the static method...

Reply:
Pushed. Let me know.

Reply:
I added few comments. We're almost there ;-). Pls squash the commits.

Reply:
Hi I think I did eveything that is needed.

Let me know.